### ISSUE TITLE
Clipboard syncing not working in 2.8.0

### ISSUE THREAD

**Issue by Omoeba** [XQuartz/XQuartz#142]
The clipboard doesn't sync unless the "Enable syncing" option is toggled off and toggled back on again. The behavior occurs with iterm2, xquartz 2.8.0, and xorg-server 1.20.10. The issue doesn't appear to occur in any of the release candidate versions, although I'm unable to find a download link to test them again.

**Comment by jeremyhu** [comment]
This has been an issue for ages.  It sometimes gets wedged for some reason, but it hasn't been triaged yet.

**Comment by Omoeba** [comment]
Yeah I've seen the issue happen in previous xquartz/xorg-server releases as well.

**Comment by jbulow** [comment]
I have never experienced this issue before upgrading to XQuartz 2.8. In 2.8 it happens very often.

**Comment by jeremyhu** [comment]
Don't know what to tell you @jbulow.  The code in 2.8 is 100% unchanged and this has been reported for years.

 * https://github.com/XQuartz/xquartz-old-tickets/blob/master/ticket/184.md
 * https://github.com/XQuartz/xquartz-old-tickets/blob/master/ticket/226.md
 * https://github.com/XQuartz/xquartz-old-tickets/blob/master/ticket/320.md
 * https://github.com/XQuartz/xquartz-old-tickets/blob/master/ticket/476.md
 * https://github.com/XQuartz/xquartz-old-tickets/blob/master/ticket/519.md
 * https://github.com/XQuartz/xquartz-old-tickets/blob/master/ticket/558.md
 * https://github.com/XQuartz/xquartz-old-tickets/blob/master/ticket/620.md
 * https://github.com/XQuartz/xquartz-old-tickets/blob/master/ticket/765.md

**Comment by jbulow** [comment]
Sorry for my confusing comment. To be more constructive:

It seems like the clipboard sync does not start correctly the first time a new version of XQuartz is installed.

**Comment by joecady** [comment]
I don't know if this is the same issue, but since updating XQuartz to 2.8.1_rc1 (xorg-server 1.20.11) I can no longer copy text from Inkscape to any app outside of XQuartz. MacOS X 10.12.6.

**Comment by CharlesARoy** [comment]
In case this can help anyone, I typed up a solution that worked for me over [on Stack Overflow](https://stackoverflow.com/questions/47822357/how-to-use-x11-forwarding-to-copy-from-vim-to-local-machine/68506108#68506108).

**Comment by YangEfei** [comment]
The same problem in XQuartz 2.8.4. 

I can get the clipboard from remote by `xclip -sel cliboard -o | pbcopy`, and the clipboard in my Mac will be instead by remote. Then I can use `command + v` to paste them. 

However, no matter I enable the `enabling sync` or not, the XQuartz can not change the clipboard automatically. So I have to set `alias xc='xclip -sel cliboard -o | pbcopy'` and run it manually before I want to paste them by `command + v`.

**Comment by ShiromMakkad** [comment]
If I run `xclip -i -selection clipboard` on the remote (centos) machine, it works. Running `xsel -b` only works once. Then I have to toggle the XQuartz clipboard sync on and off.  I don't have either `xsel` or `xclip` on my Mac just XQuartz. 

So I think it's an issue of `xsel` and many other linux programs putting copies into an incorrect register. Reading the settings of XQuartz, it seems as though there's two registers, CLIPBOARD and PRIMARY. Maybe some programs are inserting into the wrong one, but toggling syncs the two together.

**Comment by od-vennre** [comment]
> The same problem in XQuartz 2.8.4.
> 
> I can get the clipboard from remote by `xclip -sel cliboard -o | pbcopy`, and the clipboard in my Mac will be instead by remote. Then I can use `command + v` to paste them.
> 
> However, no matter I enable the `enabling sync` or not, the XQuartz can not change the clipboard automatically. So I have to set `alias xc='xclip -sel cliboard -o | pbcopy'` and run it manually before I want to paste them by `command + v`.

I do not understand this workaround and the linked commits.

Where do you run `xclip -sel cliboard -o | pbcopy`? On your Mac OS X? But it does not have `xclip`. On the remote? But it does not have `pbcopy`.

I am okay with using a workaround because toggling the Pastebin option in the GUI is becoming too annoying. I would like to understand how to use this workaround to manually sync the clipboard between my Mac OS X and a remote Linux system.

**Comment by robohack** [comment]
You use `xclip` with `pbcopy` by installing [`xclip`](https://github.com/astrand/xclip) (e.g. from https://github.com/astrand/xclip) on your macos machine, then run the whole command pipeline in an xterm running locally in/with Xquartz on the mac.

Typically I've found I only have to use `pbcopy` to get things into the macos "pastebin" -- i.e. the syncing only stops working from XQuartz to macos.  Selections copied from macos typically always appear in the XQuartz clipboard and can always be pasted to X11 apps.

I set up my window manager (I run XQuartz only in fullscreen mode and use `ctwm`) with a menu button that opens an xterm running pbcopy so that I can paste into that window in X11, then switch back to macos and paste into an app there.  That way I can see what I'm copying, though I guess I could use `tee` with `xclip` to see what it is copying.
